Wizard_chess.py

```diff
@@ -1,728 +1,308 @@
-"13":[1.95,0.65],
-"23":[1.95,1.3],
-"33":[1.95,1.95],
-"43":[1.95,2.6],
-"53":[1.95,3.25],
-"63":[1.95,3.9],
-"73":[1.95,4.55],
-"04":[2.6,0],
-"14":[2.6,0.65],
-"24":[2.6,1.3],
-"34":[2.6,1.95],
-"44":[2.6,2.6],
-"54":[2.6,3.25],
-"64":[2.6,3.9],
-"74":[2.6,4.55],
-"05":[3.25,0],
-"15":[3.25,0.65],
-"25":[3.25,1.3],
-"35":[3.25,1.95],
-"45":[3.25,2.6],
-"55":[3.25,3.25],
-"65":[3.25,3.9],
-"75":[3.25,4.55],
-"06":[3.9,0],
-"16":[3.9,0.65],
-"26":[3.9,1.3],
-"36":[3.9,1.95],
-"46":[3.9,2.6],
-"56":[3.9,3.25],
-"66":[3.9,3.9],
-"76":[3.9,4.55],
-"07":[4.55,0],
-"17":[4.55,0.65],
-"27":[4.55,1.3],
-"37":[4.55,1.95],
-"47":[4.55,2.6],
-"57":[4.55,3.25],
-"67":[4.55,3.9],
-"77":[4.55,4.55],
-}
-
-class piece():
-    pieces = {'white':{},'black':{}}
-    pieces_taken = {'white':{},'black':{}}
-    def __init__(self,piece_type,colour):
-        self.piece_type=piece_type
-        self.colour=colour
-        self.no_of_moves = 0
-        self.position = '00'
-        self.moves_list = []
-
-        if not self.piece_type in piece.pieces[self.colour]:
-            piece.pieces[self.colour][self.piece_type] = [self]
+
+class Magnet:
+    """
+    Simple wrapper around GPIO calls, just to make it more clear what's g oin
+    """
+
+    @staticmethod
+    def initialize():
+        """
+        Sets up the magnet
+        :return: none
+        """
+        GPIO.setup(7, GPIO.OUT)
+
+    @staticmethod
+    def grab():
+        """
+        Grab a piece with the magnet
+        :return: none
+        """
+        GPIO.output(7, 0)
+
+    @staticmethod
+    def release():
+        """
+        Release the piece from the magnet
+        :return: none
+        """
+        GPIO.output(7, 1)
+
+    @staticmethod
+    def cleanup():
+        """
+        Cleans up the magnet
+        :return: none
+        """
+        GPIO.cleanup()
+
+
+class MotorMover:
+    """
+    Handles moving the motor to a specific position on the board
+    """
+
+    def __init__(self, motor_x, motor_y, unit_dist=0.65):
+        """
+        Initialize the motor mover
+        :param motor_x: motor for the x axis
+        :param motor_y: motor for the y axis
+        :param unit_dist: distance that traveling 1 unit takes
+        """
+        self.x = 0
+        self.y = 0
+        self.motors = [motor_x, motor_y]
+        self.unit_dist = unit_dist
+
+        for motor in self.motors:
+            motor.init()
+
+    def _compute_dist_and_rot_dir(self, new_value, old_value):
+        """
+        Calculates the distance and rotation that the motor has to travel for going to a new location
+        :param new_value: the new location
+        :param old_value: the old location
+        :return: (distance, rotation_direction), where rotation_direction is either Motor.CLOCKWISE or Motor.ANTICLOCKWISE
+        """
+        distance = self.unit_dist * abs(old_value - new_value)
+        rotation_direction = Motor.CLOCKWISE if new_value > old_value else Motor.ANTICLOCKWISE

```

Comment:
I'm not quite sure if these rotation directions are correct, so if the pieces are moving the opposite of what they should be, its probably this line.